experimental/yarpl/examples/FlowableExamples.cpp

```diff
@@ -1,70 +1,61 @@
 // Copyright 2004-present Facebook. All Rights Reserved.
 
-#include "examples/FlowableExamples.h"
+#include "FlowableExamples.h"
+#include <thread>
+#include "reactivestreams/ReactiveStreams.h"
+#include "yarpl/Flowable.h"
+
+using namespace reactivestreams_yarpl;
+using namespace yarpl::flowable;
+
+using namespace reactivestreams_yarpl;
 
 void FlowableExamples::run() {
   std::cout << "---------------FlowableExamples::run-----------------"
             << std::endl;
 
-  class MySubscriber : public ASubscriber<int> {
-    std::unique_ptr<ASubscription> theSubscription;
-
+  class MySubscriber : public Subscriber<long> {
    public:
-    MySubscriber() {
-      std::cout << "MySubscriber CREATED" << std::endl;
+    void onSubscribe(Subscription* subscription) {
+      s_ = subscription;
+      requested = 10;
+      s_->request(10);
     }
-    ~MySubscriber() {
-      std::cout << "MySubscriber DESTROYED" << std::endl;
-    }
-    void onNext(const int& value) {
-      std::cout << "  onNext received " << value << std::endl;
+
+    void onNext(const long& t) {
+      acceptAndRequestMoreIfNecessary();
+      std::cout << "onNext& " << t << std::endl;
     }
-    void onError(const std::exception& e) {}
-    void onComplete() {}
-    void onSubscribe(std::unique_ptr<ASubscription> s) {
-      theSubscription = std::move(s);
-      theSubscription->request(10);
+
+    void onNext(long&& t) {
+      acceptAndRequestMoreIfNecessary();
+      std::cout << "onNext&& " << t << std::endl;
     }
-  };
 
-  class MySubscription : public ASubscription {
-    std::weak_ptr<ASubscriber<int>> s;
+    void onComplete() {
+      std::cout << "onComplete " << std::endl;
+    }
 
-   public:
-    ~MySubscription() {
-      std::cout << "MySubscription DESTROYED" << std::endl;
+    void onError(const std::exception_ptr error) {
+      std::cout << "onError " << std::endl;
     }
-    MySubscription(std::weak_ptr<ASubscriber<int>> _s) : s(std::move(_s)) {
-      std::cout << "MySubscription CREATED" << std::endl;
-    };
-    void cancel() {}
-    void request(uint64_t n) {
-      auto sp = s.lock();
-      if (sp) {
-        // do stuff
-        sp->onNext(1);
+
+   private:
+    Subscription* s_;
```

Comment:
nit: the member variables typically belongs to the bottom under all private methods